pkg/test/framework/label/labels.go

```diff
@@ -28,13 +28,17 @@ const (
 
 	// Multicluster indicates that the test requires a multicluster configuration.
 	Multicluster Instance = "multicluster"
+
+	// ExternalIstiod indicates that the test should be run under externalIstiod set up
+	ExternalIstiod Instance = "externalistiod"
```

Comment:
I don't think we need to add this. We should be able to infer this by looking at the config topology.. if a cluster points to a cluster other than itself we're in external istiod mode. 

See RequiresSingleCluster, that would be a simpler pattern to follow.

Reply:
Thanks @stevenctl  for the comments! I want a way to let the prow job select related tests. Now I am using label as in https://github.com/istio/test-infra/pull/2952/

env:
      - name: TEST_SELECT
        value: "-postsubmit,-flaky,+externalistiod"

any better way to control this?

Reply:
How are you selecting the topology json? I'd imagine we'd want something like the --topology flag to the integ-test-kind script

Reply:
I agree with @stevenctl here. We want to just pass in the `configTopology` flag to handle this. External istiod is just an abstract concept that manifests by setting things up a particular way. We shouldn't see anything calling out `external istiod` here at all.

The prow job is the thing that selects the topology jason, so it should also be responsible for properly setting the network, control plane, and config topology flags.

And it looks like this is already available: https://github.com/istio/istio/blob/9c7842669edde9fc697158ce6645e519a0904759/prow/integ-suite-kind.sh#L146, which ends up getting turned into the `--configTopology` flag here: https://github.com/istio/istio/blob/b2d7cacaa2fd3535fc98dc070400d11e9cb8512e/tests/integration/tests.mk#L69.